### ISSUE TITLE
HTTP/2 supporting

### ISSUE THREAD

**Issue by boyxuper** [openresty/drizzle-nginx-module#26]
Since lua_nginx_module been support HTTP/2: openresty/lua-nginx-module#566
Is there any chance we could also get drizzle-nginx-module to work with HTTP/2 ?

**Comment by agentzh** [comment]
@boyxuper You should try it out yourself. I think it should just work. I've never tested it though.

**Comment by agentzh** [comment]
@boyxuper BTW, ngx_lua does not fully support HTTP/2 yet.

**Comment by boyxuper** [comment]
@agentzh  Thx for the fast reply. I've tried drizzle with HTTP/2 but failed, and here's the error log: 

2016/04/10 02:23:11 [alert] 5834#0: *436835 epoll_ctl(1, 43) failed (17: File exists), client: 127.0.0.1, server: , request: "GET /item/245136127 HTTP/2.0", subrequest: "/_read_mysql", host: "127.0.0.1"

I'm afraid that I'm not so familiar with drizzle-nginx-module code, but I'd like to try, what should I do to fix this?

**Comment by agentzh** [comment]
@boyxuper I guess we should migrate the following two code snippets guarded by `NGX_HTTP_V2` in `src/http/ngx_http_upstream.c` of the latest NGINX core to ngx_drizzle's `src/ngx_http_drizzle_upstream.c` file:

```
src/http/ngx_http_upstream.c
490:#if (NGX_HTTP_V2)
1176:#if (NGX_HTTP_V2)
```

**Comment by jmealo** [comment]
@boyxuper @agentzh https://github.com/openresty/lua-nginx-module/issues/252#issuecomment-24369065 I'm not sure if this is related.

**Comment by agentzh** [comment]
@jmealo Yeah, the error message is similar. But I cannot derive more from just the error message itself. Patches welcome :)

**Comment by adrhc** [comment]
it doesn't work with HTTP/2:

> 2024/09/12 21:03:01 [alert] 1863890#0: *14 epoll_ctl(1, 26) failed (17: File exists), client: 192.168.0.2, server: adrhc.go.ro, request: "GET /albums/mysql/image/1000 HTTP/2.0", host: "adrhc.go.ro"
